code/parse/sexp.cpp

```diff
@@ -13547,6 +13548,23 @@ void sexp_set_subspace_drive(int node)
 	sexp_deal_with_ship_flag(CDR(node), true, Object::Object_Flags::NUM_VALUES, Ship::Ship_Flags::No_subspace_drive, Mission::Parse_Object_Flags::NUM_VALUES, set_flag);
 }
 
+//forward declarations
+extern void ai_set_preferred_primary_weapon(char *subject_name, int weapon_idx, char *target_name);
+extern void ai_clear_preferred_primary(char *subject_name, char *target_name);
+
+void sexp_good_primary_time(int node)
+{
+	char *subject_name = CTEXT(node);
```

Comment:
Done.  I moved the definition of `object_ship_wing_point_team` and the associated `#defines` to `sexp.h` in order to make it visible to `aicode.cpp`

Reply:
I should have been more explicit.  Put the `object_ship_wing_point_team` stuff back where it was, since that should really be confined to the SEXP code.  When passing to the ai code, you can use your ship/wing/team enumeration.

Reply:
In other words, use oswpt for parsing, then get the ship/wing/team information out of the oswpt struct before you pass it to aicode.